okapi-core/src/main/java/org/folio/okapi/util/MetricsHelper.java

```diff
@@ -0,0 +1,216 @@
+package org.folio.okapi.util;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.Timer.Sample;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.vertx.core.VertxOptions;
+import io.vertx.micrometer.MicrometerMetricsOptions;
+import io.vertx.micrometer.VertxInfluxDbOptions;
+import io.vertx.micrometer.backends.BackendRegistries;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.bean.ModuleInstance;
+import org.folio.okapi.common.OkapiLogger;
+
+/**
+ * Metrics handling.
+ */
+public class MetricsHelper {
+
+  private static final Logger logger = OkapiLogger.get();
+
+  private static final String METRICS_PREFIX = "org.folio.okapi";
+  private static final String METRICS_HTTP = METRICS_PREFIX + ".http";
+  private static final String METRICS_HTTP_SERVER = METRICS_HTTP + ".server";
+  private static final String METRICS_HTTP_CLIENT = METRICS_HTTP + ".client";
+  private static final String METRICS_HTTP_SERVER_PROCESSING_TIME = METRICS_HTTP_SERVER
+      + ".processingTime";
+  private static final String METRICS_HTTP_CLIENT_RESPONSE_TIME = METRICS_HTTP_CLIENT
+      + ".responseTime";
+  private static final String METRICS_HTTP_CLIENT_ERRORS = METRICS_HTTP_CLIENT
+      + ".errors";
+
+  private static final String TAG_HOST = "host";
+  private static final String TAG_TENANT = "tenant";
+  private static final String TAG_CODE = "code";
+  private static final String TAG_METHOD = "method";
+  private static final String TAG_MODULE = "module";
+  private static final String TAG_URL = "url";
+  private static final String TAG_PHASE = "phase";
+  private static final String TAG_EMPTY = "null";
+
+  static final String HOST_UNKNOWN = "unknown";
+
+  private static boolean enabled = false;
+  private static MeterRegistry registry;
+
+  private MetricsHelper() {
+  }
+
+  /**
+   * Config metrics options - specifically use InfluxDb micrometer options.
+   *
+   * @param vertxOptions   - {@link VertxOptions}
+   * @param influxUrl      - default to http://localhost:8086
+   * @param influxDbName   - default to okapi
+   * @param influxUserName - default to null
+   * @param influxPassword - default to null
+   */
+  public static void config(VertxOptions vertxOptions, String influxUrl,
+      String influxDbName, String influxUserName, String influxPassword) {
+    VertxInfluxDbOptions influxDbOptions = new VertxInfluxDbOptions()
+        .setEnabled(true)
+        .setUri(Optional.ofNullable(influxUrl).orElse("http://localhost:8086"))
```

Comment:
There is no need to create an Optional for the sole purpose of providing a default value for null.
There exist several idioms for this:
* String.toString(influxUrl, "http://localhost:8086")
* Object.toString(influxUrl, "http://localhost:8086")
* StringUtils.defaultString(influxUrl, "http://localhost:8086")
* ObjectUtils.firstNonNull(influxUrl, "http://localhost:8086")

(There are 5 such Optional instances in this file.)

Reply:
Thanks for the suggestions. For the four simple string assignments, I switched to use ternary operator to check null. That seems more intuitive. I kept Optional for the registry assignment so we do not have to depend on 3rd party libraries. I am not familiar with the first item on your list `String.toString(a, b)`, which utility library provides that?